flink-dstl/flink-dstl-dfs/src/main/java/org/apache/flink/changelog/fs/FsStateChangelogWriter.java

```diff
@@ -394,18 +394,14 @@ public void confirm(SequenceNumber from, SequenceNumber to, long checkpointId) {
                 .forEach(
                         localHandle -> {
                             changelogRegistry.stopTracking(localHandle);
-                            localChangelogRegistry.register(localHandle, checkpointId);
                         });
-    }
-
-    @Override
-    public void subsume(long checkpointId) {
         localChangelogRegistry.discardUpToCheckpoint(checkpointId);
     }
 
     @Override
     public void reset(SequenceNumber from, SequenceNumber to, long checkpointId) {
-        localChangelogRegistry.prune(checkpointId);
+        // delete all accumulated local dstl files when abort
+        localChangelogRegistry.discardUpToCheckpoint(checkpointId + 1);
```

Comment:
I don't think the `#discardUpToCheckpoint` has the same meaning as the previous `#prune`. 
`#prune` would only delete files with the specific checkpoint id, while `discardUpToCheckpoint` would delete files with checkpoint id smaller than the given one. And the `notifyCheckpointAbort` should tell the TM only abort files with that checkpoint id.

Reply:
As we discuss offline, because changelog state backend is incremental, when a checkpoint is aborted, all subsequent checkpoints that depend on this checkpoint are incomplete, It is equivalent to all previous checkpoints that are no longer useful. Therefore, we broaden the responsibility of `#abort`/`#prune` and delete all the previous useless local checkpoints.